### ISSUE TITLE
Bitmovin and exoplayer conflicting dependencies in latest version

### ISSUE THREAD

**Issue by harshmittal29** [bitmovin/bitmovin-analytics-collector-android#23]
I am using both Exoplayer and Bitmovin Analytics libraries in my project

    compile "com.google.android.exoplayer:exoplayer:$rootConfiguration.exoPlayerVersion"
    compile "com.bitmovin.analytics:collector:$rootConfiguration.bitmovinVersion"

exoPlayerVersion = '2.8.4'
 bitmovinVersion = '1.3.8'

When I run my app it gives me this error

`AGPBI: {"kind":"error","text":"Program type already present: `com.google.android.exoplayer2.source.hls.BuildConfig","sources":[{}],"tool":"D8"}

How can I exclude Exoplayer files from Bitmovin?
P.S. It was working fine until version 1.2.0

**Comment by Tigraine** [comment]
Hi @harshmittal29 - thanks for reporting this.
Yes this is a known problem at the moment as the analytics collector library supports both ExoPlayer and Bitmovin Player so we link against both versions in the gradle.

We are planning to separate the two libraries so there will be one package for the bitmovin-player and one for Exoplayer.

Can you help me understand the impact this issue has for you right now so we can prioritize accordingly?
I'll try to confirm timelines on when we can make that change.

**Comment by harshmittal29** [comment]
We have updated our Exoplayer to 2.8.4. When we tried to build Bitmovin 1.2.0 it dint work out so had to update to 1.3.8. On updating to 1.3.8 it was showing above error.
I also tried excluding Exoplayer modules

compile ("com.bitmovin.analytics:collector:$rootConfiguration.bitmovinVersion"){
 
        exclude module: 'exoplayer-smoothstreaming'
        exclude module: 'exoplayer-hls'
        exclude module: 'exoplayer-dash'
        exclude module: 'exoplayer-core'
        exclude module: 'exoplayer-ui'
    }

But then while building a release app, Proguard started giving some errors.
Our app is in production and we need to role out new version soon, so it is critical to our business.

**Comment by Tigraine** [comment]
Are you using Analytics only to monitor the Bitmovin player or also the ExoPlayer served Videos?

**Comment by harshmittal29** [comment]
Only Analytics part. We pretty much this code
           [ VideoModule.getInstance().getBitmovinAnalyticsConfig().setVideoId("" + "");    // overmind id
            VideoModule.getInstance().getBitmovinAnalyticsConfig().setCustomUserId("");  // auth id
            VideoModule.getInstance().getBitmovinAnalyticsConfig().setExperimentName("");
            VideoModule.getInstance().getBitmovinAnalyticsConfig().setCustomData1("");  // asset id
            VideoModule.getInstance().getBitmovinAnalyticsConfig().setCustomData2("");  // quest id
            VideoModule.getInstance().getBitmovinAnalyticsConfig().setCustomData3("");  //content id
            VideoModule.getInstance().getAnalyticsCollector().attachPlayer(player);

**Comment by Tigraine** [comment]
Ok understood - we'll try to release a new build of the collector this week that addresses this issue.

**Comment by Tigraine** [comment]
Hi @harshmittal29 we just released version 1.4.1 of Bitmovin Analytics that solves the issue.
We seperated out the bitmovin analytics collector into two libraries to be included in projects. One for exoplayer that's compatible with 2.8.+ and one for the bitmovin-player which avoids the issue of overlapping dependencies.

Please reference `implementation 'com.bitmovin.analytics:collector-exoplayer:1.4.1'` from your gradle build file. The setup code remains largely the same - it's just now a more specific `ExoPlayerCollector` that gets instantiated instead of `AnalyticsCollector` before.

```
ExoPlayerCollector bitmovinAnalytics = new ExoPlayerCollector(bitmovinAnalyticsConfig);
bitmovinAnalytics.attachPlayer(player);
```

Please let me know if you have any questions

**Comment by harshmittal29** [comment]
Works fine. Thanks @Tigraine
